packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -328,6 +347,78 @@ function WorklistDataGrid(props) {
+
+    // We are subtracting 1 to accommodate for the column header in the grid.
+    const rowIdsToSelect = rows.slice(selectionStartRowIndex - 1, selectionEndRowIndex).map(row => row.id);
+
+    // Determine if there are rows that are no longer in range that need to be unselected.
+    let rowIdsToUnselect = [];
+    if (rangeInfo.anchorRow > rangeInfo.previousSelectionEndRow) {
+      // The range extends upward from the anchor row
+      if (newEndRowIndex > rangeInfo.previousSelectionEndRow) {
+        // The range was moved down towards the anchor so rows that no longer qualify for the range need to be unselected.
+        rowIdsToUnselect = rows.slice(rangeInfo.previousSelectionEndRow - 1, Math.min(newEndRowIndex, rangeInfo.anchorRow)).map(row => row.id);
+      }
+    } else if (rangeInfo.anchorRow < rangeInfo.previousSelectionEndRow) {
+      // The range extends downward from the anchor row
+      if (newEndRowIndex < rangeInfo.previousSelectionEndRow) {
+        // The range was moved up towards the anchor so rows that no longer qualify for the range need to be unselected. New endRangeIndex becomes ordered End
+        rowIdsToUnselect = rows.slice(Math.max(rangeInfo.anchorRow, newEndRowIndex), rangeInfo.previousSelectionEndRow).map(row => row.id);
+      }
+    }
+
+    const ariaMessage = `${intl.formatMessage({
+      id: hasSelectableRows
+        ? 'Terra.worklist-data-grid.row-selection-multiple-rows-selected'
+        : 'Terra.worklist-data-grid.row-selection-mode-enabled',
+    })}. ${intl.formatMessage({
+      id: rowIdsToSelect.length === 1
+        ? 'Terra.worklist-data-grid.row-selection-template'
+        : 'Terra.worklist-data-grid.row-selection-selected-rows-range',
+    }, { row: rangeInfo.anchorRow, endRow: newEndRowIndex })}`;
+
+    setAriaLiveMessage(ariaMessage);
+    onRowSelect(rowIdsToSelect, rowIdsToUnselect);
+  }, [hasSelectableRows, intl, onEnableRowSelection, onRowSelect, rows]);
+
+  const selectRows = useCallback((rowId, rowIndex) => {
+    let isSelectAction = true;
+    let msgId = 'Terra.worklist-data-grid.all-rows-selected';
+    // Reset last selected row when all rows are selected.
+    rowSelection.current.rangeUsingLastIndividuallySelectedRow = null;
+
+    isSelectAction = !rows[rowIndex - 1].isSelected; // Determine if this is select or unselected.
+    // Remember the last selected row
+    rowSelection.current.rangeUsingLastIndividuallySelectedRow = { anchorRow: rowIndex };
+    if (isSelectAction) {
+      msgId = 'Terra.worklist-data-grid.row-selection-template';
+    } else {
+      msgId = 'Terra.worklist-data-grid.row-selection-cleared-template';
+    }
+    const rowLabel = rows[rowIndex - 1].ariaLabel || (rowIndex + 1);
+
+    setAriaLiveMessage(intl.formatMessage({ id: msgId }, { row: rowLabel }));
+    if (onRowSelect) {
+      if (isSelectAction) {
+        onRowSelect([rowId]);
+      } else {
+        onRowSelect([], [rowId]);
```

Comment:
The selection state should not change how we call the callback.

Reply:
As we discussed offline, not making further change.